### PR TITLE
Extend async search keep alive

### PR DESCRIPTION
**Pull request by dnhatn**
There can be a race between two GET async search requests, and the one with a lower keep_alive parameter wins the race. This scenario is not desirable as we should retain the search result for all requests. This commit ensures the keep_alive is extended and never goes backward.

Relates https://github.com/elastic/kibana/issues/88599

### CODE REVIEW

File: x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java [hunk 1]
```diff
@@ -128,7 +129,7 @@ Listener getSearchProgressActionListener() {
      */
     @Override
     public void setExpirationTime(long expirationTimeMillis) {
```
Comment (jimczi) [inline x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java:131]: Maybe change the name to reflect the new behavior (extendExpirationTimeIfNeeded) ?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java [hunk 2]
```diff
@@ -202,10 +208,12 @@ public void updateResponse(String docId,
     public void updateExpirationTime(String docId,
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java:208]: Can you update the javadocs and maybe change the name of the function (extendExpirationTimeIfNeeded) ?

**Review by jimczi — APPROVED** [review]
I left two minor comments regarding naming but the change looks great, thanks @dnhatn !

File: x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java (hunk 1 above)
Comment (dnhatn) [reply x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java]: ++, I renamed this in https://github.com/elastic/elasticsearch/pull/67877/commits/369cdc1748079ea0cd33c7301212d8ae5009d614.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java (hunk 2 above)
Comment (dnhatn) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java]: I adjusted this in https://github.com/elastic/elasticsearch/pull/67877/commits/369cdc1748079ea0cd33c7301212d8ae5009d614

**Comment by dnhatn** [comment]
Thanks Jim!

**Comment by lizozom** [comment]
@jimczi I thought we ended up agreeing that following https://github.com/elastic/kibana/pull/89570 we actually don't need this restriction.
Do you think it's justified to keep it still? Or maybe it's worth reverting?

**Comment by jimczi** [comment]
Good call @lizozom , we added the restriction before the design change in https://github.com/elastic/kibana/pull/89570 but +1 to revert. I'll open a new PR.

**Comment by jimczi** [comment]
Reverted by https://github.com/elastic/elasticsearch/pull/67877

**Pull request merged**